code/modules/antagonists/horror/horror.dm

```diff
@@ -0,0 +1,855 @@
+				H.Activate()
+			if(invisible)
+				var/datum/action/innate/horror/H = has_ability(/datum/action/innate/horror/chameleon)
+				H.Activate()
+			Update_Invisibility_Button()
+			..()
+
+/mob/living/simple_animal/horror/ex_act()
+	if(victim)
+		return
+
+	..()
+
+/mob/living/simple_animal/horror/proc/infect_victim()
+	if(!can_use_ability())
+		return
+	if(victim)
+		to_chat(src, span_warning("You are already within a host."))
+
+	if(stat == DEAD)
+		return
+
+	var/list/choices = list()
+	for(var/mob/living/carbon/C in view(1,src))
+		if(C!=src && Adjacent(C))
+			choices += C
+
+	if(!choices.len)
+		return
+	var/mob/living/carbon/C = choices.len > 1 ? input(src,"Who do you wish to infest?") in null|choices : choices[1]
+	if(!C || !src || !Adjacent(C))
+		return
+
+	if(C.has_horror_inside())
+		to_chat(src, span_warning("[C] is already infested!"))
+		return
+
+	to_chat(src, span_warning("You slither your tentacles up [C] and begin probing at their ear canal..."))
+	if(!do_mob(src, C, 30))
+		to_chat(src, span_warning("As [C] moves away, you are dislodged and fall to the ground."))
+		return
+
+	if(!C || !src)
+		return
+
+	Infect(C)
+
+/mob/living/simple_animal/horror/proc/Infect(mob/living/carbon/C)
+	if(!C)
+		return
+	var/obj/item/bodypart/head/head = C.get_bodypart(BODY_ZONE_HEAD)
+	if(!head)
+		to_chat(src, span_warning("[C] doesn't have a head!"))
+		return
+	var/hasbrain = FALSE
+	for(var/obj/item/organ/brain/X in C.internal_organs)
+		hasbrain = TRUE
+		break
+	if(!hasbrain)
+		to_chat(src, span_warning("[C] doesn't have a brain!"))
+		return
+
+	if(C.has_horror_inside())
+		to_chat(src, span_warning("[C] is already infested!"))
+		return
+	if(C.stat == DEAD) //hey look something's entered your head, maybe they'll try to revive you
+		victim.grab_ghost(force = TRUE)
+
+	if((!C.key || !C.mind) && target && C != target.current)
+		to_chat(src, span_warning("[C]'s mind seems unresponsive. Try someone else!"))
+		return
+
+	invisible = FALSE
+	Update_Invisibility_Button()
+	victim = C
+	forceMove(victim)
+	RefreshAbilities()
+	log_game("[src]/([src.ckey]) has infested [victim]/([victim.ckey]")
+
+/mob/living/simple_animal/horror/proc/secrete_chemicals()
+	if(!victim)
+		to_chat(src, span_warning("You are not inside a host body."))
+		return
+
+	if(!can_use_ability())
+		return
+
+	var content = ""
+	content += "<p>Chemicals: <span id='chemicals'>[chemicals]</span></p>"
+	content += "<table>"
+
+	for(var/datum in horror_chems)
+		var/datum/horror_chem/C = new datum()
+		if(C.chemname)
+			content += "<tr><td><a class='chem-select' href='?_src_=\ref[src];src=\ref[src];horror_use_chem=[C.chemname]'>[C.chemname] ([C.chemuse])</a><p>[C.chem_desc]</p></td></tr>"
+
+	content += "</table>"
+
+	var/html = get_html_template(content)
+
+	usr << browse(null, "window=ViewHorror\ref[src]Chems;size=600x800")
+	usr << browse(html, "window=ViewHorror\ref[src]Chems;size=600x800")
```

Comment:
DM reference is saying that you don't need to close and re-open it if you have the same window for both.
Can you verify that you've tried without the browse(null)?